app/controllers/gitlab/webhook_base_controller.rb

```diff
@@ -0,0 +1,27 @@
+module Gitlab
+  class WebhookBaseController < ActionController::Base
+    before_action :validate_secret_key
+
+    def validate_secret_key
+      head :unprocessable_entity unless valid_secret_key?
+    end
+
+    private
+
+    def valid_secret_key?
+      request.headers['X-Gitlab-Token'] == integration.data['secret_key']
+    end
+
+    def integration
+      @integration ||= project.integrations.find_by(kind: 'gitlab')
```

Comment:
Note for a followup refactor: let's extract this string to a constant (and do the same for the other integrations).

Reply:
Should I change this now?

Reply:
Since there are other integrations I think we can do it after merging that to keep the PR focused